src/runner/browser-job.js

```diff
@@ -27,13 +25,13 @@ export default class BrowserJob extends EventEmitter {
 
         this.completionQueue = [];
 
-        this.connectionErrorListener = error => this._setResult(RESULT.errored, error);
+        this.connectionErrorListener = async error => await this._setResult(RESULT.errored, error);
```

Comment:
It won't this way. Even if `connectionErrorListener` is async, its results won't be awaited because it's used as an event listener below. Node.js doesn't awaits async functions in error listeners.